addons/l10n_de/models/res_company.py

```diff
@@ -1,11 +1,31 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models, fields
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+import stdnum.de.stnr
+import stdnum.exceptions
 
 
 class ResCompany(models.Model):
     _inherit = 'res.company'
 
     l10n_de_stnr = fields.Char(string="St.-Nr.", help="Steuernummer. Scheme: ??FF0BBBUUUUP, e.g.: 2893081508152 https://de.wikipedia.org/wiki/Steuernummer")
+    l10n_de_stnr_national = fields.Char(string="Unique national SteuerNummer", compute='_compute_l10n_de_stnr_national')
     l10n_de_widnr = fields.Char(string="W-IdNr.", help="Wirtschafts-Identifikationsnummer.")
+
+    @api.depends('l10n_de_stnr')
+    @api.constrains('l10n_de_stnr')
+    def _compute_l10n_de_stnr_national(self):
```

Comment:
It's a compute function *and* a constraint ? Seems very exotic :parrot:  :thinking: It's weird to raise like that in the middle of compute functions. Wouldn't it be better to distinguish the compute from the constraint ? Those don't have the same responsibilities.

And that being said, maybe we just need the constraint, and not a computed field (it's used in a single place and not shown anywhere), keeping an explicit call to get the proper number in the export of the tax report. You can make the report and the constraint call a single function if you want, to avoid code duplication.